torch/csrc/jit/frontend/source_range.h

```diff
@@ -13,7 +13,8 @@ class SourceRangeUnpickler;
 struct SourceRange;
 
 // Source represents a code segment. It keeps track of:
-//  - text : the text of the code segment
+//  - text or text_view : the text of the code segment, or a view into it
```

Comment:
mmm this makes ownership a little tangled, right? We do not really enforce that the source text sticks around, which makes this change unsafe in general; I don't really love "maybe-owned" style classes when they're not absolutely necessary.

Possibly: we could create a `SourceView` class that shares most implementation with `Source` but operates over a view, and use that during the startup schema parsing. 

If that's too hard we definitely *could* leave as is and just say it's up to the caller to keep things straight, but ideally we can be more explicit.

Reply:
>  I don't really love "maybe-owned" style classes when they're not absolutely necessary.

I would support enabling `c10::MaybeOwned<std::string>` where the borrow type is `c10::string_view` if this turns out to be a good use case for it!